app/controllers/api/v1/jwt_controller.rb

```diff
@@ -1,12 +1,26 @@
 class API::V1::JwtController < API::APIController
 
-  # POST api/v1/jwt/firebase
+  # POST api/v1/jwt/firebase as a logged in user, or
+  # GET  api/v1/jwt/firebase?firebase_app=abc with a valid bearer token
   def firebase
-    return error('You must be logged in to use this endpoint') if !current_user
-    return error('Missing firebase_app parameter') if params[:firebase_app].empty?
+    header = request.headers["Authorization"]
+    if header && header =~ /^Bearer (.*)$/
+      token = $1
+      grant = AccessGrant.find_by_access_token(token)
+
+      return error('Cannot find AccessGrant for token #{token}') if !grant
+      return error('AccessGrant has expired') if grant.access_token_expires_at < Time.now
+
+      user = grant.user
+    else
+      return error('You must be logged in to use this endpoint') if current_visitor.anonymous?
+      user = current_visitor
```

Comment:
I think @scytacki doesn't want us to use current_visitor anymore

Reply:
Yes I'm trying to phase out current_visitor.  current_user is what you should use.  current_user will be nil if the is no user logged in.  current_visitor will always be set to something.

Reply:
:+1: Ok, that's changed. I was having trouble with current_user before, but that actually had to do with how I was passing credentials for a CORS request.